src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,7 +1,81 @@
-import React from 'react';
+import cn from 'classnames';
+import React, { useState } from 'react';
+import { getMovie } from '../../api';
+import { Movie } from '../../types/Movie';
+// import { Movie } from '../../types/Movie';
+import { MovieData } from '../../types/MovieData';
+import { MovieCard } from '../MovieCard';
 import './FindMovie.scss';
 
-export const FindMovie: React.FC = () => {
+type Props = {
+  addMovie: (movie: Movie) => void;
+};
+
+export const FindMovie: React.FC<Props> = ({ addMovie }) => {
+  const [findedMovie, setFindedMovie] = useState<Movie | null>(null);
+  const [query, setQuery] = useState('');
+  const [loading, setLoading] = useState(false);
+  const [hasError, setHasError] = useState(false);
+
+  const applyQuery = (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    setHasError(false);
+    setQuery(event.target.value);
+  };
+
+  const clearForm = () => {
+    setQuery('');
+    setFindedMovie(null);
+  };
+
+  const handleAddMovie = () => {
+    if (findedMovie) {
+      addMovie(findedMovie);
+    }
+
+    clearForm();
+  };
+
+  const findMovie = async (
+    event: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+  ) => {
+    event.preventDefault();
+    setLoading(true);
+
+    try {
+      const movieFromApi = await getMovie(query);
+
+      if ('Error' in movieFromApi) {
+        setHasError(true);
+        setLoading(false);
+
+        return;
+      }
```

Comment:
can we throw the error in the api function and catch the error in the catch {} structure?

Reply:
I hope I understood correctly ^-^

Reply:
agree, it will be better solution